### PR TITLE
Remove unroll and convert its use cases to use forEach instead

### PR DESCRIPTION
**Pull request by LMS007**
The diff is long, but straight forward, everything just got a forEach more less exactly how the test case was already written.

Only major change was to moderation-banner-test.js where I added the test description text into the test name as it felt appropriate to do so. Previously it was only in a comment.

### CODE REVIEW

File: src/sidebar/components/test/annotation-test.js [hunk 1]
```diff
@@ -917,9 +915,39 @@ describe('annotation', function() {
...
+          expected: true,
+        },
+        {
+          case_: 'the draft is shared and the group is restricted',
+          draft: draftFixtures.shared,
+          group: groupFixtures.restricted,
+          expected: true,
+        },
+      ].forEach(testCase => {
+        it(`returns #expected if ${testCase.case_}`, () => {
```
Comment (robertknight) [inline src/sidebar/components/test/annotation-test.js]: `#expected` here needs to be replaced with `${expected}`

File: src/sidebar/test/markdown-commands-test.js [hunk 2]
```diff
@@ -105,55 +104,47 @@ describe('markdown commands', function() {
         input: '<sel></sel>',
         output: '> <sel></sel>',
       },
-    ];
-
-    unroll(
-      '#tag',
-      function(fixture) {
+    ].forEach(fixture => {
+      it(`${fixture.tag}`, () => {
```
Comment (robertknight) [inline src/sidebar/test/markdown-commands-test.js]: ``it(`${fixture.tag}`,`` can be simplified to just `it(fixture.tag,`

File: src/sidebar/components/test/annotation-test.js (hunk 1 above)
Comment (LMS007) [reply src/sidebar/components/test/annotation-test.js]: good catch!

**Comment by LMS007** [comment]
> Looks like @robertknight gave a solid once-over on this, so it LGTM after his comments are addressed.
> 
> One note I'd make is that it might be nice to have a more fully-worded commit for this work. The commit summary certain explains the change clearly, but the motivation for the change might be nice as a body of the commit. This is a nice-to-have, though, not critical.
> 
> Thanks for taking on this housekeeping task. Always nice to keep our tech debt slim and things moving in the right direction!

noted, will do!

File: src/sidebar/test/markdown-commands-test.js (hunk 2 above)
Comment (LMS007) [reply src/sidebar/test/markdown-commands-test.js]: lol @ myself for this one :)

**Pull request merged**